Check install steps

Follow the installation steps from https://actinia.mundialis.de/tutorial/installation.html and document deprecated information in this ticket or directly in the documentation.
Attention: the documentation changed from rst files to md: https://github.com/mundialis/actinia_core/blob/main/docs/docs/installation.md

So I got up to https://github.com/mundialis/actinia_core/blob/main/docs/docs/installation.md?plain=1#L80-L85 after following dependencies where met:

```
apt install zlib1g-dev unzip build-essentials libbz2-dev \
  libzstd-dev gdal-bin libgdal-dev libfftw3-dev libcairo2-dev
```

However the next step is causing the following error. 

```
root@li1502-8:/tmp/grass_master/temporal# cd t.rast.sample/
root@li1502-8:/tmp/grass_master/temporal/t.rast.sample# make
if [ "/tmp/grass_master/dist.x86_64-pc-linux-gnu/scripts/t.rast.sample" != "" ] ; then GISRC=/tmp/grass_master/dist.x86_64-pc-linux-gnu/demolocation/.grassrc78 GISBASE=/tmp/grass_master/dist.x86_64-pc-linux-gnu PATH="/tmp/grass_master/dist.x86_64-pc-linux-gnu/bin:/tmp/grass_master/dist.x86_64-pc-linux-gnu/bin:/tmp/grass_master/dist.x86_64-pc-linux-gnu/scripts:$PATH" PYTHONPATH="/tmp/grass_master/dist.x86_64-pc-linux-gnu/etc/python:/tmp/grass_master/dist.x86_64-pc-linux-gnu/gui/wxpython:$PYTHONPATH" LD_LIBRARY_PATH="/tmp/grass_master/dist.x86_64-pc-linux-gnu/bin:/tmp/grass_master/dist.x86_64-pc-linux-gnu/bin:/tmp/grass_master/dist.x86_64-pc-linux-gnu/scripts:/tmp/grass_master/dist.x86_64-pc-linux-gnu/lib:/tmp/grass_master/dist.x86_64-pc-linux-gnu/lib:/usr/local/lib" LC_ALL=C LANG=C LANGUAGE=C /tmp/grass_master/dist.x86_64-pc-linux-gnu/scripts/t.rast.sample --html-description < /dev/null | grep -v '</body>\|</html>' > t.rast.sample.tmp.html ; fi
Traceback (most recent call last):
  File "/tmp/grass_master/dist.x86_64-pc-linux-gnu/scripts/t.rast.sample", line 81, in <module>
    import grass.script as gscript
  File "/tmp/grass_master/dist.x86_64-pc-linux-gnu/etc/python/grass/__init__.py", line 4, in <module>
    import six
ImportError: No module named six
make: *** [../..//include/Make/Html.make:14: t.rast.sample.tmp.html] Error 1
rm t.rast.sample.tmp.html
```

I assume it needs python. Its not clear to me which python version should be installed. When I try python3 any pip3 packages aren't picked up. So for example `apt install python3-debian python3-pip && pip3 install six` doesn't fix the issue.

python2 however does seem to work but fails for another reason which seems to be python2 incompatible package.


```bash
apt install -y curl && \
  curl https://bootstrap.pypa.io/pip/2.7/get-pip.py --output get-pip.py && \
  python2 get-pip.py

pip2 install six

# ...

  File "/tmp/grass_master/dist.x86_64-pc-linux-gnu/etc/python/grass/pygrass/vector/basic.py", line 9, in <module>
    from collections.abc import Iterable
ImportError: No module named abc
make: *** [../..//include/Make/Html.make:14: t.rast.sample.tmp.html] Error 1
rm t.rast.sample.tmp.html
```

looking into this `from collections.abc import Iterable` seems to be that this is compatible with python3. See: https://stackoverflow.com/questions/55882715/difference-between-from-collections-import-container-and-from-collections-abc-im

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -20,7 +20,7 @@ and GRASS GIS git version and compile it.
 ```
 
 
-2.  Set Python 3 environemtn
+2.  Set Python 3 environment
 
 ```bash
  # recommended to give Python3 precedence over Python2 (which is end-of-life since 2019)

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -9,7 +9,7 @@ These installation instructions are based on a Ubuntu Linux system (other
 operating systems and distributions are fine, too).
 
 
-First install PPROJ and GRASS GIS by downloading the latest packages
+First install PROJ and GRASS GIS by downloading the latest packages
 and GRASS GIS git version and compile it.
 
 1.  Install the latest projection library
@@ -21,6 +21,8 @@ and GRASS GIS git version and compile it.
 
 
 2.  Set Python 3 environment
+
+Note that this step changes `python` to point to `python3`, system-wide:
 
 ```bash
  # recommended to give Python3 precedence over Python2 (which is end-of-life since 2019)
@@ -40,13 +42,10 @@ and GRASS GIS git version and compile it.
 
  cd grass78
 
- export INTEL="-std=gnu99 -fexceptions -fstack-protector -m64"
- export MYCFLAGS="-Wall -ggdb -fno-fast-math -fno-common $INTEL $MYGCC"
- export MYLDFLAGS="-Wl,--no-undefined"
- export LD_LIBRARY_PATH="/usr/local/lib"
- export LDFLAGS="$MYLDFLAGS"
- export CFLAGS="$MYCFLAGS"
- export CXXFLAGS="$MYCXXFLAGS"
+ # production related compilation flags (no debugging enabled)
+ export CFLAGS="-O2 -std=gnu99 -fexceptions -fstack-protector -m64"
+ export CXXFLAGS="-O2"
+ export LDFLAGS="-Wl,--no-undefined -Wl,-z,now"
 
  ./configure \
    --with-cxx \

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -15,8 +15,8 @@ and GRASS GIS git version and compile it.
 1.  Install the latest projection library
 
 ```bash
- sudo apt-get update
- sudo apt-get install proj-bin proj-data
+sudo apt-get update
+sudo apt-get install proj-bin proj-data
 ```
 
 
@@ -25,143 +25,143 @@ and GRASS GIS git version and compile it.
 Note that this step changes `python` to point to `python3`, system-wide:
 
 ```bash
- # recommended to give Python3 precedence over Python2 (which is end-of-life since 2019)
- sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+# recommended to give Python3 precedence over Python2 (which is end-of-life since 2019)
+sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 1
 ```
 
 
 3.  Compile and install GRASS GIS and additional modules:
 
 ```bash
- cd /tmp
+cd /tmp
 
- wget https://grass.osgeo.org/grass78/source/snapshot/grass-7.8.git_src_snapshot_latest.tar.gz
- tar xzvf grass-7.8.git_src_snapshot_latest.tar.gz
- mv grass-7.8.git_src_snapshot_????_??_?? grass78
- rm -f grass-7.8.git_src_snapshot_latest.tar.gz
+wget https://grass.osgeo.org/grass78/source/snapshot/grass-7.8.git_src_snapshot_latest.tar.gz
+tar xzvf grass-7.8.git_src_snapshot_latest.tar.gz
+mv grass-7.8.git_src_snapshot_????_??_?? grass78
+rm -f grass-7.8.git_src_snapshot_latest.tar.gz
 
- cd grass78
+cd grass78
 
- # production related compilation flags (no debugging enabled)
- export CFLAGS="-O2 -std=gnu99 -fexceptions -fstack-protector -m64"
- export CXXFLAGS="-O2"
- export LDFLAGS="-Wl,--no-undefined -Wl,-z,now"
+# production related compilation flags (no debugging enabled)
+export CFLAGS="-O2 -std=gnu99 -fexceptions -fstack-protector -m64"
+export CXXFLAGS="-O2"
+export LDFLAGS="-Wl,--no-undefined -Wl,-z,now"
 
- ./configure \
-   --with-cxx \
-   --enable-largefile \
-   --with-proj-share=/usr/share/proj \
-   --with-gdal \
-   --with-geos \
-   --with-sqlite \
-   --with-cairo --with-cairo-ldflags=-lfontconfig \
-   --with-fftw \
-   --with-netcdf \
-   --with-bzlib \
-   --with-zstd \
-   --without-postgres \
-   --without-freetype \
-   --without-openmp \
-   --without-opengl \
-   --without-nls \
-   --without-mysql \
-   --without-odbc \
-   --without-openmp \
-   --prefix=/usr/local
+./configure \
+  --with-cxx \
+  --enable-largefile \
+  --with-proj-share=/usr/share/proj \
+  --with-gdal \
+  --with-geos \
+  --with-sqlite \
+  --with-cairo --with-cairo-ldflags=-lfontconfig \
+  --with-fftw \
+  --with-netcdf \
+  --with-bzlib \
+  --with-zstd \
+  --without-postgres \
+  --without-freetype \
+  --without-openmp \
+  --without-opengl \
+  --without-nls \
+  --without-mysql \
+  --without-odbc \
+  --without-openmp \
+  --prefix=/usr/local
 
- make -j8
- sudo make install
+make -j8
+sudo make install
 
- cd temporal/
- git clone https://github.com/mundialis/t.rast.sample
- cd t.rast.sample
- make
- cd ../..
- sudo make install
+cd temporal/
+git clone https://github.com/mundialis/t.rast.sample
+cd t.rast.sample
+make
+cd ../..
+sudo make install
 
- # t.rast.udf
- # see https://github.com/mundialis/openeo-addons
+# t.rast.udf
+# see https://github.com/mundialis/openeo-addons
 
- cd display/
- git clone https://github.com/mundialis/d_rast_multi d.rast.multi
- cd d.rast.multi
- make
- cd ../..
- sudo make install
+cd display/
+git clone https://github.com/mundialis/d_rast_multi d.rast.multi
+cd d.rast.multi
+make
+cd ../..
+sudo make install
 ```
 
 
 4.  Download the test datasets ("locations") and place them into a
-    specific directory that will be used by actinia as a persistent database:
+   specific directory that will be used by actinia as a persistent database:
 
 ```bash
- mkdir -p $HOME/actinia/grassdb
- cd $HOME/actinia/grassdb
- wget https://grass.osgeo.org/sampledata/north_carolina/nc_spm_08_grass7.tar.gz
- tar xzvf nc_spm_08_grass7.tar.gz
- rm -f nc_spm_08_grass7.tar.gz
- mv nc_spm_08_grass7 nc_spm_08
+mkdir -p $HOME/actinia/grassdb
+cd $HOME/actinia/grassdb
+wget https://grass.osgeo.org/sampledata/north_carolina/nc_spm_08_grass7.tar.gz
+tar xzvf nc_spm_08_grass7.tar.gz
+rm -f nc_spm_08_grass7.tar.gz
+mv nc_spm_08_grass7 nc_spm_08
 
- wget https://storage.googleapis.com/datentransfer/ECAD.tar.gz
- tar xzvf ECAD.tar.gz
- rm -f ECAD.tar.gz
+wget https://storage.googleapis.com/datentransfer/ECAD.tar.gz
+tar xzvf ECAD.tar.gz
+rm -f ECAD.tar.gz
 
- wget https://storage.googleapis.com/datentransfer/LL.tar.gz
- tar xzvf LL.tar.gz
- rm -f LL.tar.gz
- mv LL latlong_wgs84
+wget https://storage.googleapis.com/datentransfer/LL.tar.gz
+tar xzvf LL.tar.gz
+rm -f LL.tar.gz
+mv LL latlong_wgs84
 ```
 
 
 5.  Actinia Core must be installed and run within a (virtual) Python3 environment:
 
 ```bash
- mkdir -p $HOME/actinia/workspace/tmp
+mkdir -p $HOME/actinia/workspace/tmp
 
- cd /tmp
+cd /tmp
 
- git clone https://github.com/mundialis/actinia_core.git actinia_core
- # virtualenv -p python3 actinia_venv
- # source actinia_venv/bin/activate
- cd actinia_core
- pip install -r requirements.txt
- python setup.py install
- # deactivate
+git clone https://github.com/mundialis/actinia_core.git actinia_core
+# virtualenv -p python3 actinia_venv
+# source actinia_venv/bin/activate
+cd actinia_core
+pip install -r requirements.txt
+python setup.py install
+# deactivate
 ```
 
 
 6.  Actinia API must be installed and run within a (virtual) Python3 environment:
 
 ```bash
- mkdir -p $HOME/actinia/workspace/tmp
+mkdir -p $HOME/actinia/workspace/tmp
 
- cd /tmp
+cd /tmp
 
- git clone https://github.com/mundialis/actinia-api.git actinia-api
- # virtualenv -p python3 actinia_venv
- # source actinia_venv/bin/activate
- cd actinia-api
- python setup.py install
- # deactivate
+git clone https://github.com/mundialis/actinia-api.git actinia-api
+# virtualenv -p python3 actinia_venv
+# source actinia_venv/bin/activate
+cd actinia-api
+python setup.py install
+# deactivate
 ```
 
 
 7.  Make sure that a redis service is running and create a user with
-    actinia-user tool
+   actinia-user tool
 
 ```bash
- cd /tmp
+cd /tmp
 
- # install and start redis server
- apt-get install -y redis
- redis-server &
+# install and start redis server
+apt-get install -y redis
+redis-server &
 
- # source actinia_venv/bin/activate
- # Create the superuser
- actinia-user create -u superadmin -w abcdefgh -r superadmin -g admin -c 100000000000 -n 1000 -t 6000
+# source actinia_venv/bin/activate
+# Create the superuser
+actinia-user create -u superadmin -w abcdefgh -r superadmin -g admin -c 100000000000 -n 1000 -t 6000
 
- # Start the actinia service with
- actinia-server --host 0.0.0.0 --port 5000
+# Start the actinia service with
+actinia-server --host 0.0.0.0 --port 5000
 ```
 
 
@@ -169,10 +169,10 @@ Note that this step changes `python` to point to `python3`, system-wide:
 
 ```bash
 ACTINIA_VERSION="v2"
- export ACTINIA_URL="http://localhost:5000/api/${ACTINIA_VERSION}"
- export AUTH='-u superadmin:abcdefgh'
+export ACTINIA_URL="http://localhost:5000/api/${ACTINIA_VERSION}"
+export AUTH='-u superadmin:abcdefgh'
 
- curl ${AUTH} -X GET ${ACTINIA_URL}/locations
+curl ${AUTH} -X GET ${ACTINIA_URL}/locations
 ```
 
 The `curl` command call should report back:

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -33,7 +33,8 @@ sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 1
 3.  Compile and install GRASS GIS and additional modules:
 
 ```bash
-cd /tmp
+# or some other location of your choice, /tmp will not keep data between reboots
+cd /tmp 
 
 # fetch weekly generated latest snapshot of GRASS GIS stable
 wget https://grass.osgeo.org/grass78/source/snapshot/grass-7.8.git_src_snapshot_latest.tar.gz

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -28,7 +28,7 @@ and GRASS GIS git version and compile it.
 ```
 
 
-3.  Conmpile and install GRASS GIS and additional modules:
+3.  Compile and install GRASS GIS and additional modules:
 
 ```bash
  cd /tmp

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -169,7 +169,7 @@ and GRASS GIS git version and compile it.
 8.  Check the service with curl
 
 ```bash
- ACTINIA_VERSION="v2"
+ACTINIA_VERSION="v2"
  export ACTINIA_URL="http://localhost:5000/api/${ACTINIA_VERSION}"
  export AUTH='-u superadmin:abcdefgh'
 

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -35,6 +35,7 @@ sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 1
 ```bash
 cd /tmp
 
+# fetch weekly generated latest snapshot of GRASS GIS stable
 wget https://grass.osgeo.org/grass78/source/snapshot/grass-7.8.git_src_snapshot_latest.tar.gz
 tar xzvf grass-7.8.git_src_snapshot_latest.tar.gz
 mv grass-7.8.git_src_snapshot_????_??_?? grass78
@@ -121,12 +122,9 @@ mkdir -p $HOME/actinia/workspace/tmp
 cd /tmp
 
 git clone https://github.com/mundialis/actinia_core.git actinia_core
-# virtualenv -p python3 actinia_venv
-# source actinia_venv/bin/activate
 cd actinia_core
 pip install -r requirements.txt
 python setup.py install
-# deactivate
 ```
 
 
@@ -138,11 +136,8 @@ mkdir -p $HOME/actinia/workspace/tmp
 cd /tmp
 
 git clone https://github.com/mundialis/actinia-api.git actinia-api
-# virtualenv -p python3 actinia_venv
-# source actinia_venv/bin/activate
 cd actinia-api
 python setup.py install
-# deactivate
 ```
 
 
@@ -156,7 +151,6 @@ cd /tmp
 apt-get install -y redis
 redis-server &
 
-# source actinia_venv/bin/activate
 # Create the superuser
 actinia-user create -u superadmin -w abcdefgh -r superadmin -g admin -c 100000000000 -n 1000 -t 6000
 
@@ -181,4 +175,4 @@ The `curl` command call should report back:
 {"locations":["nc_spm_08","ECAD","latlong_wgs84"],"status":"success"}
 ```
 
-Success!
+Success and welcome to actinia!

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -5,41 +5,40 @@ A local installation requires a full GRASS GIS installation. Make sure
 all requirements are fulfilled to compile GRASS GIS from the git
 repository.
 
-First install Proj4 and GRASS GIS by downloading the latest git version
-and compile it.
+These installation instructions a based on a Ubuntu Linux system (other
+operating systems and distributions are fine, too).
 
-1.  Install the latest projection library.
+
+First install PPROJ and GRASS GIS by downloading the latest packages
+and GRASS GIS git version and compile it.
+
+1.  Install the latest projection library
 
 ```bash
- cd /tmp
-
- wget http://download.osgeo.org/proj/proj-4.9.3.tar.gz
-
- tar xzvf proj-4.9.3.tar.gz
-
- cd proj-4.9.3/nad
-
- wget http://download.osgeo.org/proj/proj-datumgrid-1.8.zip
-
- unzip proj-datumgrid-1.8.zip
-
- cd ..
-
- ./configure
- make -j4
- sudo make install
+ sudo apt-get update
+ sudo apt-get install proj-bin proj-data
 ```
 
-2.  Install GRASS GIS and additional modules:
+
+2.  Set Python 3 environemtn
+
+```bash
+ # recommended to give Python3 precedence over Python2 (which is end-of-life since 2019)
+ sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+```
+
+
+3.  Conmpile and install GRASS GIS and additional modules:
 
 ```bash
  cd /tmp
 
  wget https://grass.osgeo.org/grass78/source/snapshot/grass-7.8.git_src_snapshot_latest.tar.gz
  tar xzvf grass-7.8.git_src_snapshot_latest.tar.gz
- mv grass-7.8.git_src_snapshot_????_??_?? grass_master
+ mv grass-7.8.git_src_snapshot_????_??_?? grass78
+ rm -f grass-7.8.git_src_snapshot_latest.tar.gz
 
- cd grass_master
+ cd grass78
 
  export INTEL="-std=gnu99 -fexceptions -fstack-protector -m64"
  export MYCFLAGS="-Wall -ggdb -fno-fast-math -fno-common $INTEL $MYGCC"
@@ -52,10 +51,8 @@ and compile it.
  ./configure \
    --with-cxx \
    --enable-largefile \
-   --with-proj=/usr/local/lib \
-   --with-proj-share=/usr/local/share/proj \
+   --with-proj-share=/usr/share/proj \
    --with-gdal \
-   --with-python \
    --with-geos \
    --with-sqlite \
    --with-cairo --with-cairo-ldflags=-lfontconfig \
@@ -71,33 +68,32 @@ and compile it.
    --without-mysql \
    --without-odbc \
    --without-openmp \
-   --without-ffmpeg \
    --prefix=/usr/local
 
- make -j16
+ make -j8
  sudo make install
 
- cd temporal
- git clone https://bitbucket.org/huhabla/temporal-raster-sampling.git t.rast.sample
+ cd temporal/
+ git clone https://github.com/mundialis/t.rast.sample
  cd t.rast.sample
  make
- sudo make install
- cd ..
-
- git clone https://github.com/huhabla/t_rast_aggr_func.git t.rast.aggr_func
- cd t.rast.aggr_func
- make
+ cd ../..
  sudo make install
 
- cd ../../display/
- git clone https://bitbucket.org/huhabla/d_rast_multi.git d.rast.multi
+ # t.rast.udf
+ # see https://github.com/mundialis/openeo-addons
+
+ cd display/
+ git clone https://github.com/mundialis/d_rast_multi d.rast.multi
  cd d.rast.multi
  make
+ cd ../..
  sudo make install
 ```
 
-3.  Download the test locations and place them into a specific directory
-    that will be used by actinia as persistent database:
+
+4.  Download the test datasets ("locations") and place them into a
+    specific directory that will be used by actinia as a persistent database:
 
 ```bash
  mkdir -p $HOME/actinia/grassdb
@@ -117,7 +113,8 @@ and compile it.
  mv LL latlong_wgs84
 ```
 
-4.  Actinia Core must be run within a virtual python3 environment:
+
+5.  Actinia Core must be installed and run within a (virtual) Python3 environment:
 
 ```bash
  mkdir -p $HOME/actinia/workspace/tmp
@@ -125,47 +122,64 @@ and compile it.
  cd /tmp
 
  git clone https://github.com/mundialis/actinia_core.git actinia_core
- virtualenv -p python3.5 actinia_venv
- source actinia_venv/bin/activate
+ # virtualenv -p python3 actinia_venv
+ # source actinia_venv/bin/activate
  cd actinia_core
  pip install -r requirements.txt
  python setup.py install
- deactivate
+ # deactivate
 ```
 
-5.  GRASS GIS 7.8 requires a Python 3 environment for running, so we
-    create one (not needed if only Python 3 is present on the machine).
-    This environment must be specified in the actinia config file:
+
+6.  Actinia API must be installed and run within a (virtual) Python3 environment:
 
 ```bash
+ mkdir -p $HOME/actinia/workspace/tmp
+
  cd /tmp
 
- virtualenv -p python2.7 grass_venv
- source grass_venv/bin/activate
- cd actinia_core
- pip install -r requirements-grass-gis.txt
+ git clone https://github.com/mundialis/actinia-api.git actinia-api
+ # virtualenv -p python3 actinia_venv
+ # source actinia_venv/bin/activate
+ cd actinia-api
+ python setup.py install
+ # deactivate
 ```
 
-6.  Make sure that a redis service is running and create a user with
+
+7.  Make sure that a redis service is running and create a user with
     actinia-user tool
 
 ```bash
  cd /tmp
- source actinia_venv/bin/activate
 
+ # install and start redis server
+ apt-get install -y redis
+ redis-server &
+
+ # source actinia_venv/bin/activate
  # Create the superuser
  actinia-user create -u superadmin -w abcdefgh -r superadmin -g admin -c 100000000000 -n 1000 -t 6000
 
  # Start the actinia service with
- actinia_server --host 0.0.0.0 --port 5000
+ actinia-server --host 0.0.0.0 --port 5000
 ```
 
-7.  Check the service with curl
+
+8.  Check the service with curl
 
 ```bash
-ACTINIA_VERSION="v2"
+ ACTINIA_VERSION="v2"
  export ACTINIA_URL="http://localhost:5000/api/${ACTINIA_VERSION}"
  export AUTH='-u superadmin:abcdefgh'
 
  curl ${AUTH} -X GET ${ACTINIA_URL}/locations
 ```
+
+The `curl` command call should report back:
+
+```bash
+{"locations":["nc_spm_08","ECAD","latlong_wgs84"],"status":"success"}
+```
+
+Success!

--- a/docs/docs/installation.md
+++ b/docs/docs/installation.md
@@ -5,7 +5,7 @@ A local installation requires a full GRASS GIS installation. Make sure
 all requirements are fulfilled to compile GRASS GIS from the git
 repository.
 
-These installation instructions a based on a Ubuntu Linux system (other
+These installation instructions are based on a Ubuntu Linux system (other
 operating systems and distributions are fine, too).
 
 

